packages/venia-ui/lib/RootComponents/Category/category.js

```diff
@@ -1,177 +1,63 @@
-                id: Number(id),
-                filters: newFilters,
-                pageSize: Number(pageSize),
-                sort: { [currentSort.sortAttribute]: currentSort.sortDirection }
-            }
-        });
-        window.scrollTo({
-            left: 0,
-            top: 0,
-            behavior: 'smooth'
-        });
-    }, [
-        currentPage,
-        currentSort,
-        filterTypeMap,
+    const talonProps = useCategory({
         id,
-        pageSize,
-        runQuery,
-        search
-    ]);
-
-    const totalPagesFromData = data
-        ? data.products.page_info.total_pages
-        : null;
-
-    useEffect(() => {
-        setTotalPages(totalPagesFromData);
-        return () => {
-            setTotalPages(null);
-        };
-    }, [setTotalPages, totalPagesFromData]);
-
-    // If we get an error after loading we should try to reset to page 1.
-    // If we continue to have errors after that, render an error message.
-    useEffect(() => {
-        if (error && !loading && currentPage !== 1) {
-            setCurrentPage(1);
+        queries: {
+            getCategory: GET_CATEGORY,
+            getFiltersIntrospection: FILTER_INTROSPECTION,
+            getPageSize: GET_PAGE_SIZE
         }
-    }, [currentPage, error, loading, setCurrentPage]);
-
-    // Reset the current page back to one (1) when the search string, filters
-    // or sort criteria change.
-    useEffect(() => {
-        // We don't want to compare page value.
-        const prevSearch = new URLSearchParams(previousSearch.current);
-        const nextSearch = new URLSearchParams(search);
-        prevSearch.delete('page');
-        nextSearch.delete('page');
+    });
+
+    const {
+        error,
+        metaDescription,
+        loading,
+        categoryData,
+        pageControl,
+        sortProps,
+        pageSize
+    } = talonProps;
 
-        if (
-            prevSearch.toString() !== nextSearch.toString() ||
-            previousSort.current.sortAttribute.toString() !==
-                currentSort.sortAttribute.toString() ||
-            previousSort.current.sortDirection.toString() !==
-                currentSort.sortDirection.toString()
-        ) {
-            // The search term changed.
-            setCurrentPage(1);
-            // And update the ref.
-            previousSearch.current = search;
-            previousSort.current = currentSort;
-        }
-    }, [currentSort, previousSearch, search, setCurrentPage]);
+    const classes = mergeClasses(defaultClasses, props.classes);
 
-    if (error && currentPage === 1 && !loading) {
+    if (error && pageControl.currentPage === 1 && !loading) {
         if (process.env.NODE_ENV !== 'production') {
             console.error(error);
         }
         return <div>Data Fetch Error</div>;
     }
 
     // Show the loading indicator until data has been fetched.
-    if (totalPagesFromData === null) {
+    if (loading) {
```

Comment:
This is a good change but I would move this `if` block above the error case block and then remove the `&& !loading`.